storage/resource_databricks_files.go

```diff
@@ -0,0 +1,127 @@
+package storage
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/md5"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/databricks/databricks-sdk-go/service/files"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/databricks/terraform-provider-databricks/workspace"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func getContentReader(data *schema.ResourceData) (io.ReadCloser, error) {
+	source := data.Get("source").(string)
+	var reader io.ReadCloser
+	var err error
+	if source != "" {
+		reader, err = os.Open(source)
+		if err != nil {
+			return nil, err
+		}
+	}
+	contentBase64 := data.Get("content_base64").(string)
+	if contentBase64 != "" {
+		decodedString, err := base64.StdEncoding.DecodeString(contentBase64)
+		if err != nil {
+			return nil, err
+		}
+		reader = io.NopCloser(bytes.NewReader(decodedString))
+		if err != nil {
+			return nil, err
+		}
+	}
+	return reader, err
+}
+
+func ResourceFiles() *schema.Resource {
```

Comment:
it should be singular to be consistent with other file resource

```suggestion
func ResourceFile() *schema.Resource {
```